pkg/api/goprobe/server/config.go

```diff
@@ -2,10 +2,80 @@ package server
 
 import (
 	"net/http"
+	"net/url"
+	"strings"
 
+	"github.com/els0r/goProbe/cmd/goProbe/config"
+	gpapi "github.com/els0r/goProbe/pkg/api/goprobe"
 	"github.com/gin-gonic/gin"
 )
 
-func (server *Server) postConfig(c *gin.Context) {
-	c.JSON(http.StatusNotImplemented, nil)
+func (server *Server) getConfig(c *gin.Context) {
+	iface := c.Param(ifaceKey)
+	ifaces := c.Request.URL.Query().Get(gpapi.IfacesQueryParam)
+
+	resp := &gpapi.ConfigResponse{}
+	resp.StatusCode = http.StatusOK
+
+	var err error
+	ifaces, err = url.QueryUnescape(ifaces)
+	if err != nil {
+		resp.StatusCode = http.StatusBadRequest
+		resp.Error = err.Error()
+
+		c.AbortWithStatusJSON(resp.StatusCode, resp)
+		return
+	}
+
+	ctx := c.Request.Context()
+
+	if iface != "" {
+		resp.Ifaces = server.captureManager.Config(ctx, iface)
+	} else {
+		if ifaces != "" {
+			// fetch all specified
+			resp.Ifaces = server.captureManager.Config(ctx, strings.Split(ifaces, ",")...)
```

Comment:
What happens if an invalid interface is provided (since there's no error handling we probably would silently ignore it, but maybe something bad might happen)? Ideally an invalid interface would cause an error which we could hand over as `HTTP 400` though...

Reply:
I dunno. Here, we are simply getting the interfaces, so if you provide iface `bubukitty`, it just won't return data for it.

For the PUT call, this is probably more relevant.